src/main/java/seedu/address/model/meeting/UniqueMeetingList.java

```diff
@@ -0,0 +1,137 @@
+package seedu.address.model.meeting;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.meeting.exceptions.DuplicateMeetingException;
+import seedu.address.model.meeting.exceptions.MeetingNotFoundException;
+
+/**
+ * A list of meetings that enforces uniqueness between its elements and does not allow nulls.
+ * A meeting is considered unique by comparing using {@code Meeting#isSameMeeting(Meeting)}. As such, adding and
+ * updating of meetings uses Meeting#isSameMeeting(Meeting) for equality so as to ensure that the meeting being added
+ * or updated is unique in terms of identity in the UniqueMeetingList. However, the removal of a meeting uses
+ * Meeting#equals(Object) so as to ensure that the meeting with exactly the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Meeting#isSameMeeting(Meeting)
+ */
+public class UniqueMeetingList implements Iterable<Meeting> {
+
+    private final ObservableList<Meeting> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Meeting> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent meeting as the given argument.
+     */
+    public boolean contains(Meeting toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameMeeting);
+    }
+
+    /**
+     * Adds a meeting to the list.
+     * The meeting must not already exist in the list.
+     */
+    public void add(Meeting toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateMeetingException();
+        }
```

Comment:
Check whether meetings clashes also

Reply:
This is resolved with the latest commit where UniqueMeetingList is completely replaced with NoConflictMeetingList.